jbs/src/main/java/org/openjdk/skara/jbs/IssueUtil.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package org.openjdk.skara.jbs;
+
+import org.openjdk.skara.issuetracker.Issue;
+
+import java.util.List;
+import java.util.Optional;
+
+public class IssueUtil {
+
+    /**
+     * Find the closest issue from the provided issue list according to the provided fix version.
+     * This method is similar to Backports#findIssue, but this method can handle all the fix versions of the issue
+     * instead of only the main fix version and can receive an issue list instead of only the primary issue.
+     *
+     * If one of the issues has the correct fix version, use it.
+     * Else, if one of the issues has a matching <N>-pool-<opt> fix version, use it.
+     * Else, if one of the issues has a matching <N>-pool fix version, use it.
+     * Else, if one of the issues has a "scratch" fix version, use it.
+     * Otherwise, return empty.
+     *
+     * A "scratch" fixVersion is empty, "tbd.*", or "unknown".
+     */
+    public static Optional<Issue> findClosestIssue(List<Issue> issueList, JdkVersion fixVersion) {
```

Comment:
> I think this method belongs in the Backports class, close to Backports#findIssue. They do almost the same thing, so if we are to maintain two similar implementations of the version matching order, they should reside close to each other.

The `Backports` class mainly handles something about backport. But this method is only about issue instead of backport. 
Actully, the method `Backports#findIssue` should delegate to a method such as `IssueUtil#findClosestIssue(List<Issue> issueList, JdkVersion fixVersion, boolean onlySearchMainFixVersion)` after getting the backport issues to find the closest issue by using the main fix version. Such refactor is good but I don't think it should be done at this patch. So I would like to keep the utilility method about issues at a seperate class so that we can do more refactor in the future. The class name `IssueUtil` may need to be adjusted to a better name.

> Could we move more of the common functionality to Backports. Something like Backports.findCsr(Issue primary, JdkVersion jdkVersion) that would encapsulate most of this repeated functionality?

I moved the method `findCsr` and `csrLink` to the class `IssueUtil`. Again, the class name `Backports` is not good to place such methods.

Reply:
From your description, now I think the `Backports` make sense to contain these methods. I moved them to `Backports` and removed the file `IssueUtil`.